packages/mdc-menu-surface/index.ts

```diff
@@ -0,0 +1,207 @@
+    this.foundation_.setQuickOpen(quickOpen);
+  }
+
+  initialSyncWithDOM() {
+    const parentEl = this.root_.parentElement;
+    const hasParentAnchor = parentEl && parentEl.classList.contains(cssClasses.ANCHOR);
+
+    this.anchorElement = hasParentAnchor ? parentEl : null;
+    this.previousFocus_ = null;
+    this.firstFocusableElement_ = null;
+    this.lastFocusableElement_ = null;
+
+    if (this.root_.classList.contains(cssClasses.FIXED)) {
+      this.setFixedPosition(true);
+    }
+
+    this.handleKeydown_ = (evt) => this.foundation_.handleKeydown(evt as KeyboardEvent);
+    this.handleBodyClick_ = (evt) => this.foundation_.handleBodyClick(evt as MouseEvent);
+
+    this.registerBodyClickListener_ = () => document.body.addEventListener('click', this.handleBodyClick_);
+    this.deregisterBodyClickListener_ = () => document.body.removeEventListener('click', this.handleBodyClick_);
+
+    this.root_.addEventListener('keydown', this.handleKeydown_);
+    this.root_.addEventListener(strings.OPENED_EVENT, this.registerBodyClickListener_);
+    this.root_.addEventListener(strings.CLOSED_EVENT, this.deregisterBodyClickListener_);
+  }
+
+  destroy() {
+    this.root_.removeEventListener('keydown', this.handleKeydown_);
+    this.root_.removeEventListener(strings.OPENED_EVENT, this.registerBodyClickListener_);
+    this.root_.removeEventListener(strings.CLOSED_EVENT, this.deregisterBodyClickListener_);
+    super.destroy();
+  }
+
+  /**
+   * Removes the menu-surface from it's current location and appends it to the
+   * body to overcome any overflow:hidden issues.
+   */
+  hoistMenuToBody() {
+    document.body.appendChild(this.root_);
+    this.setIsHoisted(true);
+  }
+
+  /** Sets the foundation to use page offsets for an positioning when the menu is hoisted to the body. */
+  setIsHoisted(isHoisted: boolean) {
+    this.foundation_.setIsHoisted(isHoisted);
+  }
+
+  /** Sets the element that the menu-surface is anchored to. */
+  setMenuSurfaceAnchorElement(element: Element) {
+    this.anchorElement = element;
+  }
+
+  /** Sets the menu-surface to position: fixed. */
+  setFixedPosition(isFixed: boolean) {
+    if (isFixed) {
+      this.root_.classList.add(cssClasses.FIXED);
+    } else {
+      this.root_.classList.remove(cssClasses.FIXED);
+    }
+
+    this.foundation_.setFixedPosition(isFixed);
+  }
+
+  /** Sets the absolute x/y position to position based on. Requires the menu to be hoisted. */
+  setAbsolutePosition(x: number, y: number) {
+    this.foundation_.setAbsolutePosition(x, y);
+    this.setIsHoisted(true);
+  }
+
+  /** @param corner Default anchor corner alignment of top-left surface corner. */
+  setAnchorCorner(corner: Corner) {
+    this.foundation_.setAnchorCorner(corner);
+  }
+
+  setAnchorMargin(margin: Partial<MenuDistance>) {
+    this.foundation_.setAnchorMargin(margin);
+  }
+
+  getDefaultFoundation(): MDCMenuSurfaceFoundation {
+    const rootEl = this.root_ as HTMLElement;
```

Comment:
if we define `root_!: HTMLElement`, then this case can be removed

Reply:
Done